h/scripts/init_db.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
```

Comment:
There's a new `<APP>/scripts/init_db.py` script.

This is a new version of the `bin/make_db` script. All DB initialization is now done with this script, whether for the dev environment, for tests (unit or functional), or for a production environment.
  
The script has a `--delete` command to delete any existing tables before creating new ones (used by the tests).
  
It has a `--create` command to create new DB tables from the models (if the DB isn't stamped by Alembic). This is used in dev, tests and production when creating a new DB.
  
And it has a `--stamp` command to stamp the DB with the latest version (if the DB isn't already stamped by Alembic). This is used by `make db` and `make dev`.
  
Usage:

```terminal
$ tox -e dev --run-command 'python3 -m h.scripts.init_db --help'
usage: init_db.py [-h] [--delete] [--create] [--stamp]

Initialize the DB.

optional arguments:
  -h, --help  show this help message and exit
  --delete    delete any existing DB tables
  --create    create new DB tables from the models
  --stamp     stamp the DB with the latest Alembic version
```
  
I think it's probably best to keep these kind of Python scripts (which may need to import dependencies from the app's environment, such as `sqlalchemy`, or even import modules from the app itself) in `<APP>`/scripts/` and keep `bin/` for standalone bash scripts because:
  
* It allows the script to be run like this:
  
      python3 -m <APP>.scripts.init_db [options]

  This has a couple of advantages:
    
  1. You know for sure that it's running the script in the `python3` interpreter's virtual environment (whatever interpreter `python3` resolves to in your current `PATH`).
  2. The command works from any directory. You don't need to know the absolute or relative path to the `init_db.py` file.
  3. An `<APP>.scripts.foo` can be either an `<APP>/scripts/foo.py` file with an `if __name__ == "__main__"` block or if we ever have more complex scripts it can be an `<APP>/scripts/foo/` package with a `__main__.py` file.

* Since the scripts are in the `<APP>` directory we can easily add unit tests for them in `tests/unit/app/scripts/` if we want to. The scripts aren't going to potentially get missed by code formatting and linting either.